src/main/scala/com/normation/cfclerk/services/impl/Cf3PromisesFileWriterServiceImpl.scala

```diff
@@ -330,11 +330,16 @@ class Cf3PromisesFileWriterServiceImpl(
     val (ncfTechniques, techniques) = policies.partition(_.providesExpectedReports)
 
     val ncfBundleSeqName = ncfTechniques.flatMap(x => x.bundlesequence.map(x =>x.name))
+
     // We mustn't quote the call NCF_REPORT_DEFINITION_BUNDLE_NAME, otherwise CFEngine will assume it refers to the complete string (with parens)
     val ncfBundleSeq     = ncfBundleSeqName.map(name => s"${NCF_REPORT_DEFINITION_BUNDLE_NAME}(${name}), ${name}").mkString(", ")
 
     val nonNcfBundleSeq  = techniques.flatMap(x => x.bundlesequence.map(x =>x.name)).mkString(", \"", "\", \"", "\"")
 
+    // We need to remove zero-length bundles from the bundlesequence (like, if there is no ncf bundles to call)
+    // to avoid having two successives commas in the bundlesequence
+    val bundleSeq = Seq[String](nonNcfBundleSeq, ncfBundleSeq).filter(_.length>0)
```

Comment:
don't you want to trim, to, so that if there is only space, the bundle is filtered out ?